contrib/babelfishpg_tsql/src/tsql_for/forxml.c

```diff
@@ -283,10 +332,33 @@ tsql_row_to_xml_path(StringInfo state, Datum record, const char *element_name, b
 		if (!isnull)
 		{
 			allnull = false;
-			appendStringInfo(state, "<%s>%s</%s>",
-							 colname,
-							 map_sql_value_to_xml_value(colval, datatype_oid, true),
-							 colname);
+			if(NameStr(att->attname)[0] == '@')
```

Comment:
Thought: Is it possible to include the changes of `validate_attribute_centric_col_names_xml` in this for loop only to avoid multiple iterations?